internal/controller/pipelinerollout_progressive.go

```diff
@@ -90,20 +92,25 @@ func (r *PipelineRolloutReconciler) processUpgradingPipelineStatus(
 		}
 	}
 
-	pipelineStatus, err := kubernetes.ParseStatus(existingUpgradingPipelineDef)
+	pipelineStatus, err := parsePipelineStatus(existingUpgradingPipelineDef)
 	if err != nil {
 		return false, fmt.Errorf("failed to parse Pipeline Status from pipeline CR: %+v, %v", existingUpgradingPipelineDef, err)
 	}
 
-	pipelinePhase := numaflowv1.PipelinePhase(pipelineStatus.Phase)
+	pipelinePhase := pipelineStatus.Phase
 	if pipelinePhase == numaflowv1.PipelinePhaseFailed {
-		err = r.updatePipelineLabel(ctx, r.restConfig, existingUpgradingPipelineDef, "")
+		// Mark the failed new pipeline recyclable.
+		err = r.updatePipelineLabel(ctx, r.restConfig, existingUpgradingPipelineDef, string(common.LabelValueUpgradeRecyclable))
```

Comment:
that's right. I forgot to pause the pipeline. I will add a todo here. so your guys can pick it up.

Reply:
its a simple change however the reason why I didn't add it, is I didn't cover the test for it.

Reply:
I do actually think that we may not want to necessarily delete the Failed pipeline, as it could be good to diagnose after the fact what happened. But then I guess somebody will need the way to perform the delete maybe through ArgoCD Action. 
Maybe this will be configurable ultimately.